### PR TITLE
NAWQAAN-585

### PR DESCRIPTION
**Pull request by jiwalker-usgs**
Here is the service code.  I updated the nar_data with extra functions and will finish that up and open a PR over there soon.

### CODE REVIEW

File: nar-webservices/src/main/java/gov/usgs/cida/nar/mybatis/model/PestSites.java [hunk 1]
```diff
@@ -0,0 +1,1432 @@
...
+	private int ndetsHh3;
+	private int ndetsAq3;
+	private int nHh3;
+	private double nNuts;
+	private double nNutsDet;
+	private double nNutsDetHh;
+	private double nNutsDetAq;
+	private int nNutHh;
+	private String plot;
+
```
Comment (cschroed-usgs) [inline nar-webservices/src/main/java/gov/usgs/cida/nar/mybatis/model/PestSites.java]: This is not a blocker, but remember that this class would stop here if we were using project Lombok, or similar.

File: nar-webservices/src/main/java/gov/usgs/cida/nar/webservice/TimeseriesWebservice.java [hunk 2]
```diff
@@ -8,6 +8,7 @@
 import gov.usgs.cida.nar.mybatis.model.Discqw;
 import gov.usgs.cida.nar.mybatis.model.Mflow;
 import gov.usgs.cida.nar.mybatis.model.Mloads;
```
Comment (cschroed-usgs) [inline nar-webservices/src/main/java/gov/usgs/cida/nar/webservice/TimeseriesWebservice.java]: Are these new imports used?

File: nar-webservices/src/main/java/gov/usgs/cida/nar/service/PestSitesService.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+		return request(siteQwId);
+	}
+
+	@Override
+	public void setSiteQwId(List<String> siteQwId) {
+		this.siteQwId = siteQwId;
+	}
+
+	@Override
+	public TimeStepDensity getTimeStepDensity() {
```
Comment (cschroed-usgs) [inline nar-webservices/src/main/java/gov/usgs/cida/nar/service/PestSitesService.java]: Should we remove the need for these unsupported dudes in some way? We had speculated that the pesticides and non-pesticides might be entirely separate APIs. Should we make the existing time series stuff implement a NARNutrientAndFlowService interface, and the pesticides a NARPesticideService interface? Could also make both interfaces implement a parent NARService interface.

File: nar-webservices/src/test/java/gov/usgs/cida/nar/service/PestSitesServiceTest.java [hunk 4]
```diff
@@ -0,0 +1,53 @@
+package gov.usgs.cida.nar.service;
+
+import gov.usgs.cida.nar.mybatis.dao.PestSitesDao;
+import gov.usgs.cida.nar.mybatis.model.PestSites;
+import java.util.List;
+import com.google.common.collect.Lists;
```
Comment (cschroed-usgs) [inline nar-webservices/src/test/java/gov/usgs/cida/nar/service/PestSitesServiceTest.java]: That Lists package is super handy.

File: nar-webservices/src/test/java/gov/usgs/cida/nar/service/PestSitesServiceTest.java [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+import static org.mockito.Mockito.when;
+
+/**
+ *
+ * @author jiwalker
+ */
+public class PestSitesServiceTest {
+	
+	private PestSitesDao mockedDao;
+	private static final String SITE_QW_ID = "100";
```
Comment (cschroed-usgs) [inline nar-webservices/src/test/java/gov/usgs/cida/nar/service/PestSitesServiceTest.java]: Why are some of the values shared across tests via constants, yet others are duplicated? I'm a fan of the constants. I can think of reasons for doing all constants, or doing none. Why is a mixture helpful?

**Comment by cschroed-usgs** [comment]
:+1: No blockers here, just some questions.

File: nar-webservices/src/test/java/gov/usgs/cida/nar/service/PestSitesServiceTest.java (hunk 5 above)
Comment (jiwalker-usgs) [inline nar-webservices/src/test/java/gov/usgs/cida/nar/service/PestSitesServiceTest.java]: No reason really, some just seemed more like constants.

File: nar-webservices/src/main/java/gov/usgs/cida/nar/service/PestSitesService.java (hunk 3 above)
Comment (jiwalker-usgs) [inline nar-webservices/src/main/java/gov/usgs/cida/nar/service/PestSitesService.java]: Let's hold off until we incorporate the other pesticide data.  It isn't clear to me what the interfaces are and how they differ at this point.

File: nar-webservices/src/test/java/gov/usgs/cida/nar/service/PestSitesServiceTest.java (hunk 5 above)
Comment (cschroed-usgs) [inline nar-webservices/src/test/java/gov/usgs/cida/nar/service/PestSitesServiceTest.java]: Roger that.

File: nar-webservices/src/main/java/gov/usgs/cida/nar/service/PestSitesService.java (hunk 3 above)
Comment (cschroed-usgs) [inline nar-webservices/src/main/java/gov/usgs/cida/nar/service/PestSitesService.java]: Sounds good!

**Pull request merged**